v3/src/components/map/v2-map-exporter.ts

```diff
@@ -0,0 +1,129 @@
+                                : 0
+  const legendCollection = legendAttributeId && dataset
+            ? dataset.getCollectionForAttribute(legendAttributeId) : undefined
+  return {
+    _links_: {
+      // shouldn't ever get here without a dataset
+      context: guidLink("DG.DataContextRecord", toV2Id(dataset?.id ?? "undefined")),
+      hiddenCases: dataConfiguration.hiddenCases.map(hiddenCase => guidLink("DG.Case", toV2Id(hiddenCase))),
+      ...maybeGuidLink("legendColl", "DG.Collection", legendCollection?.id),
+      ...maybeGuidLink("legendAttr", "DG.Attribute", attributeDescriptions.legend?.attributeID),
+    },
+    isVisible: isVisible ?? false,
+    legendRole: 0,
+    legendAttributeType
+  }
+}
+
+export const v2MapExporter: V2TileExportFn = ({ tile }) => {
+  const { name, _title, content } = tile
+  const mapContent = isMapContentModel(content) ? content : undefined
+
+  if (!mapContent) return
+  const v2LayerModels: Array<ICodapV2MapLayerStorage> = []
+
+  mapContent.layers.forEach((layer) => {
+    const { dataConfiguration: { dataset } } = layer
+
+    if (dataset) {
+      if (layer.type === kMapPointLayerType) {
+        const {displayItemDescription, gridModel, pointsAreVisible, connectingLinesAreVisible} = layer
+        const {itemColor, itemStrokeColor, pointSizeMultiplier} = displayItemDescription
+
+        const pointLayerModel: ICodapV2MapPointLayerStorage = {
+          ...exportMapBaseLayerStorage(layer),
+          pointColor: itemColor ?? "",
+          strokeColor: itemStrokeColor ?? "",
+          pointSizeMultiplier: pointSizeMultiplier ?? 1,
+          // TODO_V2_EXPORT: transparency and strokeTransparency
+          transparency: 1,
+          strokeTransparency: 1,
+          pointsShouldBeVisible: pointsAreVisible ?? false,
+          grid: {
+            gridMultiplier: gridModel.gridMultiplier ?? 1,
+            isVisible: gridModel.isVisible ?? false,
+          },
+          connectingLines: {
+            isVisible: connectingLinesAreVisible || false,
+            // TODO_V2_EXPORT: enableMeasuresForSelection
+            enableMeasuresForSelection: false
+          }
+          // TODO_V2_EXPORT: linesShouldBeVisible
+        }
+        v2LayerModels.push(pointLayerModel)
+      } else if (layer.type === kMapPolygonLayerType) {
+        const {polygonDescription} = layer
+
+        const polygonLayerModel: ICodapV2MapPolygonLayerStorage = {
+          ...exportMapBaseLayerStorage(layer),
+          areaColor: polygonDescription._itemColors[0] ?? "",
+          // TODO_V2_EXPORT: areaTransparency
+          areaTransparency: 1,
+          areaStrokeColor: polygonDescription.itemStrokeColor ?? "",
+          // TODO_V2_EXPORT: areaStrokeTransparency
+          areaStrokeTransparency: 1,
+          strokeSameAsFill: polygonDescription.itemStrokeSameAsFill ?? false
+        }
+        // V2 needs polygon layers appear before the point layers
+        v2LayerModels.unshift(polygonLayerModel)
```

Comment:
As near as I can tell from looking at the v2 code, point layer models are written out before polygon layer models. I think this needs further investigation before proceeding down this path.